File: frontend/src/types/events.ts
/** System events DTO (Spec 008 / Spec 009 §5). */

export type SystemEventSeverity = "info" | "warning" | "critical";

export type SystemEventDTO = {
  event_id: string;
  tick_id: number;
  event_type: string;
  display_code: string;
  severity: SystemEventSeverity;
  message: string;
  payload?: Record<string, unknown>;
};

export type SystemEventsResponse = {
  events: SystemEventDTO[];
};

File: src/market_abm/api/schemas/events.py
# Назначение файла: DTO system_events для REST/WS (Slice 8.3).
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class SystemEventDTO(BaseModel):
    event_id: str
    tick_id: int
    event_type: str
    display_code: str
    severity: Literal["info", "warning", "critical"]
    message: str
    payload: dict[str, object] = Field(default_factory=dict)


class SystemEventsResponse(BaseModel):
    events: list[SystemEventDTO]
